lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -211,14 +211,24 @@ namespace Js
         while (true)
         {
             const PropertyRecord *currentPropertyRecord = GetTypePath()->GetPropertyIdUnchecked(currentSlotIndex);
-            currentType = currentTypeHandler->PromoteType<false>(currentType, PathTypeSuccessorKey(currentPropertyRecord->GetPropertyId(), currentAttributes), false, scriptContext, instance, &currentSlotIndex);
+            PropertyIndex index;
+            currentType = currentTypeHandler->PromoteType<false>(currentType, PathTypeSuccessorKey(currentPropertyRecord->GetPropertyId(), currentAttributes), false, scriptContext, instance, &index);
```

Comment:
>index [](start = 192, length = 5)

Call to InitializePath below was using the `currentSlotIndex`  updated in PromoteType. Was that a bug all along?

Reply:
IIRC, the difference between currentSlotIndex and this local "index" only becomes possible when we start supporting accessors here, because of the extra slot we make for the setter.